bin/ch-test

```diff
@@ -331,19 +356,29 @@ if [[ ! -d $tmp_ ]]; then
     chmod 700 "$tmp_"
 fi
 
-# Determine source build vs prefix installation.
-if [[ -d ${ch_base}/test ]]; then
-    # source build
-    export CHTEST_DIR=${ch_base}/test
+# Find test directories. (At one point we tried to unify the paths between the
+# two conditions using a deeper directory hierarchy and symlinks to . in the
+# source code to keep it from being even deeper, but this became too unruly.)
+if [[ -d ${ch_base}/share ]]; then
+    # installed
+    CHTEST_DIR=${ch_base}/libexec/charliecloud/test
+    CHTEST_EXAMPLES_DIR=${ch_base}/share/doc/charliecloud/examples
```

Comment:
I'm not sure I agree that the examples directory should be in`${prefix}/share`. The FHS states that /usr/share is for read-only architecture independent data files. Examples has executable scripts used by `ch-test` and bat files required by `bats`, both of which are architecture dependent (linux).

IMO, it belongs in libexec with our `test` directory.

Reply:
I'm pretty sure "architecture independent" refers to CPU architecture. FHS is Linux specific and Charliecloud is Linux specific. And everything in `examples` can be read-only; I didn't check the actual permissions bits. So IMO it meets those requirements.

On a quick skim of FHS 3.0 for `/usr/share`, I don't see anything disallowing executables. On my Debian box, there's actually quite a lot of executable files there:

```
$ find /usr/share -executable -type f | wc -l
652
```

I think there's value having the examples in `share/doc` instead of `libexec` because they're a lot easier for the user to find. `chtest/Build` in particular shows how to build an image with no builder.

That said, I did wonder about the `.bats` files. While they do describe very precisely how we expect the example images to work, they require non-trivial knowledge of the test suite to understand. They could be moved to a parallel hierarchy in `test/examples`. It would complicate matching examples to their Bats files a bit. That is why I didn't do it already.